Makefile

```diff
@@ -17,20 +19,29 @@ check: check_tools get_vendor_deps
 ########################################
 ### Build
 
+# build and install build a binary inside docker
 build:
-	go build $(BUILD_FLAGS) -o build/tendermint ./cmd/tendermint/
+	@XC_OS='$(GOOS)' XC_ARCH='$(GOARCH)' BUILD_TAGS='$(BUILD_TAGS)' sh -c "'$(CURDIR)/scripts/dist.sh'"
+	@cp ./build/pkg/$(GOOS)_$(GOARCH)/tendermint ./build
```

Comment:
If we keep the old simple behavior, then dist.sh could call `make build`.

Logically, the dependency should be dist --(depends on)--> build, but this change flips that.  This will cause problems as the codebase evolves.  For example, now it becomes difficult to add certain post-build dist-only behavior in dist.sh, where it belongs.  Please revert.

Reply:
I think the intention was to have the dist pathway use the same tooling as the build pathway, so that as long as developers can ensure that build works, then it's more likely that dist works, perhaps then satisfying CI.  Was that the intent?

I think it's better to keep build/dist separate, and to keep build for development (or production if you want to compile your own binary and not rely on our distributions).  Dist can use build, or not.  Either way, I think this separation is better than merging them together.

I think the original problem that Greg mentioned was that CI was broken due to a change in our make task structure.  The solution is to ensure that those who break CI are responsible for fixing it... not to make it less likely to break CI.

Reply:
I agree, the build should not depend on dist. dist should call `make build`. build / packaging / distribution is three separate process, let's keep their codebase separate. The `Makefile` is responsible for building. CI automation is responsible for packaging, because packaging should only be possible after tests finished. I don't mind if packaging is yet another `Makefile` target, as long as I can run it against a tested binary.